Initial introduction documentation

e-hirsch opened the pull request:
I've created a high level overview over EOS to enable new interested user to quickly understand what EOS is all about and also what not. Hope this helps a bit. If anybody has additional ideas or suggestions I would be glad to discuss this in the Discussions section.

If it's okay for you I will keep adding and changing documentation and its structure over the next weeks so it would be good to keep the feature/documentation branch open.

e-hirsch commented:
Seems that the pre-commit checks fails due to a missing end-of-file newline in the provided svg files. Maybe we can exclude them from this check as the files are created by the drawing tool and you normally won't have any influence to change this behaviour.

b0661 reviewed (changes requested):
Fine icons. Good figures. Please adapt to already described architecture. Did you check the logos and icons to be free to use?

e-hirsch commented:
> Fine icons. Good figures. Please adapt to already described architecture. Did you check the logos and icons to be free to use?

Yes, I have checked that the icons are free to use. They are the material design icons from pictogrammers.com and [are distributed under Apache License 2.0](https://pictogrammers.com/docs/general/license/).

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,115 @@
+
+Access to the data and functions of the components can be done in a variety of ways. Node-RED offers a large number of types of nodes that allow access via the protocols commonly used in this area, such as Modbus or MQTT. Access to any existing databases, such as InfluxDB or PostgreSQL, is also possible via nodes provided by Node-RED.
+
+It becomes easier if a smart home solution like Homa Assistant, openHAB or ioBroker or solutions such as evcc or openWB are already in use. In this case, these solutions already take over the technical integration and communication with the components at a technical level and Node-RED offers nodes for accessing these solutions, so that the corresponding sources can be easily integrated into a flo ...
```
Okay, thought that would be clear with the sentence "One possible solution...". But you are right, I could make that even clearer by listing some other and I will also change the diagram to reflect other solutions.
A link is also fine, I had saved that for the next iteration.

e-hirsch replied:
I've added Home Assistant in the text and also in the diagram

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,115 @@
+- Charge level the electric car should reach in the next few hours
+- Consumers to run in the next few hours
+
+There are various service providers available for PV forecasting that calculate forecast data for a PV system based on the various influencing factors, such as system size, orientation, location, time of year and weather conditions. EOS also offers a simple PV forecasting service which can be used. This service uses public data in the background.
```
I had saved adding meaningful links for the next iteration.

e-hirsch replied:
Link added

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,115 @@
+
+If required, the simulation result can also be created and downloaded in graphical form as a PDF from EOS.
+
+## Integration
```
I will reference the direct integration via Home Assistant. The Add-on seems to use the "RESTful Command" integration and an automation.

e-hirsch replied:
Added an additional paragraph below the smart home solution paragraph with a hint of a possible solution and Duettings Add-on.

b0661 commented:
> And just to be sure - should I adapt the already described architecture diagrams to my style or my diagrams to the style of the other architecture diagrams or add home assistant and the EOS dashboard? I would go for the first one.

You may adapt to your style. 

We had a lot of changes in the feature/config-nested branch that will become the next main. So better you use this branch to add your changes to.

In the feature/config-nested branch there is an "intermediate" logo and icon that we use. It is here: https://github.com/Akkudoktor-EOS/EOS/tree/feature/config-nested/src/akkudoktoreos/server/dash/assets. We use the Aukkudoktor.net logo as long as nobody steps up to provide an own EOS logo.

Regarding the EOF failures that you get. This should be automatically corrected. Add the changed files afterwards.

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,115 @@
+
+### Input Data
+
+![Overview](../_static/introduction/overview.svg)
```
Added HA and reworked a little bit.

e-hirsch commented:
Maybe we can add the changes to both branches? I would like to make the new introductory documentation available to the interested public sooner rather than later. The information could be overlooked in the config-nested branch.

b0661 commented:
> Maybe we can add the changes to both branches? I would like to make the new introductory documentation available to the interested public sooner rather than later. The information could be overlooked in the config-nested branch.

Feel free to provider pull requests to main and feature/config-nested.

b0661 commented:
The black Icons are black in dark mode also, which makes them unreadable. Please choose something that also works in dark mode.

e-hirsch commented:
> The black Icons are black in dark mode also, which makes them unreadable. Please choose something that also works in dark mode.

Who uses dark mode ;-). I already had this fear. That's difficult with transparent background and would be a bunch of work. 
The new version of diagrams that I've pushed have a white background. The current template has a light grey background which works with the white background of the diagrams. It will also work with a dark background even if it doesn't look quite as nice anymore.

e-hirsch commented:
I've checked the [intro-page](https://akkudoktor-eos.readthedocs.io/en/feature-documentation/akkudoktoreos/introduction.html) with Safari, Chrome and Firefox on my Laptop and on my mobile. Can't see any issues. I've attached a screenshot how it looks like on my mobile.
![EOS-Intro-Screenshot](https://github.com/user-attachments/assets/01510edc-7ae5-40e4-b8da-39f07c85f746)

e-hirsch commented:
I've played around with using png instead of svg which seems to solve the issue with GitHub. I've changed all diagrams to png and the image references in the document. In this context, I noticed that the pre-commit fails because of the line length in the document and corrected this also.

e-hirsch commented:
> > Maybe we can add the changes to both branches? I would like to make the new introductory documentation available to the interested public sooner rather than later. The information could be overlooked in the config-nested branch.
> 
> Feel free to provider pull requests to main and feature/config-nested.

I will create another pull request if we got this one completed

e-hirsch commented:
I've also changed the index.md file to reflect the proposal from [this discussion](https://github.com/Akkudoktor-EOS/EOS/discussions/440#discussioncomment-12330874) ... on how we could re-structure the documentation for future expansion.

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+![Integration](../_static/introduction/integration.png)
+
+However, the components are not integrated by the EOS itself, but must be intergrated by
+the owner using an integration component and currently requires some effort and technical
```
the owner -> the user

e-hirsch replied:
Fine with it

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+the owner using an integration component and currently requires some effort and technical
+know-how.
+
+Any [integration](#integration-page) solution that can act as an intermediary between the
```
"integration solution" - why not just "integration" ? What is the difference between integration and integration solution?

e-hirsch replied:
The difference is that the integration solution is the piece of software that enables you to create the integration between two endpoints. One integration is the path from one endpoint to the other with the messages flowing through adapters, message filters, splitters, aggregators, enrichers, routers, translators,... you name it. In Apache Camel, a route is a synonym for integration. This is why I would differentiate between those two terms.

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+know-how.
+
+Any [integration](#integration-page) solution that can act as an intermediary between the
+components and the REST API of EOS can be used. One possible solution that enables the
```
"solution" ? Node Red and home assistant are home automation systems. Solution is very generic and I think for an introduction we should be as specific as possible to avoid misunderstandings.

e-hirsch replied:
Node-RED may be used as a home automation component but is an automation solution that can be used for a wide range of automation problems. Node-RED calls itself a low-code application builder... seems to be fashionable at the moment to call everything low-code. From my point of view it's an integration solution in its core. There are flowing messages from endpoint A to endpoint B. And on its way the message(s) are transformed and routed, etc.

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+Any [integration](#integration-page) solution that can act as an intermediary between the
+components and the REST API of EOS can be used. One possible solution that enables the
+integration of components and EOS is Node-RED. Another solution could be Home Assistant
+itself.
```
"itself" can only be understood if one knows that Node Red can also be integrated into Home Assistant. I would assume that the "usual" reader of an introduction does not know that.

e-hirsch replied:
You won't need to know or have Node-RED integrated into Home Assistant. What I wanted to express here is that you are able to use EOS with standard mechanisms of Home Assistant (lines 147-149). The "Duetting"-Add-on only uses the "RESTful Command" integration and an automation from Home Assistant.

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+commonly used in this area, such as Modbus or MQTT. Access to any existing databases,
+such as InfluxDB or PostgreSQL, is also possible via nodes provided by Node-RED.
+
+It becomes easier if a smart home solution like Homa Assistant, openHAB or ioBroker or
+solutions such as evcc or openWB are already in use. In this case, these solutions
+already take over the technical integration and communication with the components at a
+technical level and Node-RED offers nodes for accessing these solutions, so that the
+corresponding sources can be easily integrated into a flow.
```
Solutions -> please be more precise.

e-hirsch replied:
Maybe "the aforementioned solutions". Wouldn't make sense for me to repeat everything from the sentence before.

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+
+In Home Assistant you could use an automation to prepare the input payload for EOS and
+then use the RESTful integration to call EOS. Based on this concept there is already a
+solutuion created by [Duetting](#duetting-solution).
```
solution -> home assistant addon

e-hirsch replied:
Fine with that one

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+then use the RESTful integration to call EOS. Based on this concept there is already a
+solutuion created by [Duetting](#duetting-solution).
+
+The plan created by EOS must also be executed via this integration component, with the
```
New term "integration component" - what is the difference to "integration solution" and "integration"?

b0661 commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+- [German Video explaining the basic concept and installation process for the early version of EOS (YouTube)](https://www.youtube.com/live/ftQULW4-1ts?si=oDdBBifCpUmiCXaY)
+- [German Forum of Akkudoktor EOS](https://akkudoktor.net/c/der-akkudoktor/eos)
+- [Akkudoktor-EOS GitHub Repository](https://github.com/Akkudoktor-EOS/EOS)
+- [Official EOS Documentation](https://akkudoktor-eos.readthedocs.io/en/latest/)
```
Delete Official - we are an open source project. You may add latest as the link points to latest.

e-hirsch replied:
Tried to make clear that this is the documentation coming from the team that created EOS and not from a fork - but I don't care about the word "official".

b0661 reviewed (changes requested):
Please squash your commits into one. That will also ease the application to the main and feature/config-nested branch.

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+There are various service providers available for PV forecasting that calculate forecast
+data for a PV system based on the various influencing factors, such as system size,
+orientation, location, time of year and weather conditions. EOS also offers a
+[simple PV forecasting service](#prediction-page) which can be used. This service uses
```
Simple in the sense of not being able to configure more complex plants the easy way. E.g. when you have an east and a west string. Maybe you can configure to plants and then need to request both forecasts and aggregate the results... feasible but not simple. Correct me if I'm wrong. If EOS does all the requests and aggregations in the background I would indeed not call it simple. Anyways - I have no problem with deleting the word simple.

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+the electricity generated by the PV system cost-effectively for electromobility.
+
+After initial experiences, the desire to include battery storage and dynamic electricity
+prices in the solution soon arises. The market already offers various commercial and
+non-commercial solutions for this, such as the popular open source hardware and software
+solutions evcc or openWB.
```
I tried to keep it short, but now I see that I need to expand on it a bit to be more understandable. 

openWB does not provide forecasts but allows to charge the car with current PV surplus. For now evcc also allows only to use current PV surplus but they are working on the feature of including predictions for an upcoming release.

e-hirsch commented on docs/akkudoktoreos/introduction.md:
```diff
@@ -0,0 +1,165 @@
+non-commercial solutions for this, such as the popular open source hardware and software
+solutions evcc or openWB.
+
+However, many solutions only take current values such as PV power output, charge level of
+the battery storage system or the current electricity price into account and/or only
+focus on specific consumers, such as the electric car.
+
+The Akkudoktor EOS takes the possibilities to a new level by taking into account the
+relevant values ​​of the upcoming hours of the generation system with battery storage as
+well as other energy-intensive consumers.
+
```
Just out of curiosity - which solutions do the same as EOS?

And if there are other soltions - why has Andreas build this solution? And why does the project team think that anyone should use this solution? What is the reward for digging into this complex solution? Why care about this project?

Answering this questions was the goal of this "marketing speech". I wouldn't delete it but giving the right answers to the interested user and potential contributors.

b0661 replied:
> Just out of curiosity - which solutions do the same as EOS?

Not the same but similar emhass, batpred, Sunny Home Manager, ...

> Answering this questions was the goal of this "marketing speech". I wouldn't delete it but giving the right answers to the interested user and potential contributors.

Ok, but please avoid buzzwords like "takes the possibilities to a new level" that do not really provide guidance nor content and are wrong compared to the existing optimization software. 

To distinct EOS from others:
- The way the optimization is done - genetic optimization based on device simulations vs. e.g. ML models (e.g. emhass).
- open source vs. commercial (e.g. Sunny Home Manager)

EOS solves the problem to schedule the loads in an optimal way.

e-hirsch replied:
I loved especially the "new level" :stuck_out_tongue_winking_eye: . It seems that marketing speech is somehow not our thing as Germans.

Anyway, I have read a bit about the similar solutions and will try to work out the differences. This will then lead to the main differentiating features of EOS.

e-hirsch commented:
> Please squash your commits into one. That will also ease the application to the main and feature/config-nested branch.

No problem, will do that after the rework of the paragraphs that need revision.

e-hirsch commented:
Finally did all the rework and squashed all the commits into one. Should now be fine for merging into main and after this was successful I will create the pull request for the feature/config-nested branch.

e-hirsch commented:
The documentation under https://akkudoktor-eos.readthedocs.io/en/latest/ still shows the "old" documentation while https://akkudoktor-eos.readthedocs.io/en/feature-documentation/ shows the newer version. I thought that with merging the pull request into main the documentation under "latest" would be automatically deployed. Is there something to do to make that happen?

b0661 commented:
@Lasall is currently preparing to merge the feature branch into main. This will hopefully also update the main documentation.
